tests/UserContainerTests.cpp

```diff
@@ -0,0 +1,68 @@
+#include <gtest/gtest.h>
+
+#include "UserContainer.hpp"
+#include "User.hpp"
+
+struct UserContainerTests : public ::testing::Test
+{
+    UserContainerTests()
+    {
+        userContainer_.createUser(User("Kamil", "Kamil.Waszkiewicz", "wieczorowa", "Kamil.Waszkiewicz", "Kamil.Waszkiewicz"));
+    }
+
+    UserContainer userContainer_;
+};
+
+TEST_F(UserContainerTests, canConstructUserContainer)
+{
+    UserContainer uc;
+}
+
+TEST_F(UserContainerTests, canAddUserToUserContainer)
+{
+    User u1("Szymon", "SzymonGajewski", "wieczorowa", "SzymonGajewski", "SzymonGajewski");
+    userContainer_.createUser(u1);
+    
+    auto result = userContainer_.retriveUserByNick("SzymonGajewski");
+    ASSERT_TRUE(result.has_value());  
+    ASSERT_TRUE(!result.value().getName().compare("Szymon"));
```

Comment:
you can implement comparison operator and just compare users.

Reply:
implemented

Reply:
Implemented